### ISSUE TITLE
`nonminimal_bool` wrongly showed the macro definition in suggestion

### ISSUE THREAD

**Issue by profetia** [rust-lang/rust-clippy#14404]
### Summary

The lint `nonminimal_bool` falsefully showed the macro's definition in its suggestion, which it should not show but the original code.

### Reproducer

I tried this code:

```rust
enum TyKind {
    Ref(i32, i32, i32),
    Other,
}

struct Expr;

fn is_mutable(expr: &Expr) -> bool {
    todo!()
}

fn should_not_give_macro(ty: TyKind, expr: Expr) {
    if !(matches!(ty, TyKind::Ref(_, _, _)) && !is_mutable(&expr)) {
        todo!()
    }
}
```

I expected to see this happen:
```log
warning: this boolean expression can be simplified
  --> src/main.rs:32:8
   |
32 |     if !(matches!(ty, TyKind::Ref(_, _, _)) && !is_mutable(&expr)) {
   |        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#nonminimal_bool
   = note: `#[warn(clippy::nonminimal_bool)]` on by default
help: try
   |
32 ~     if !matches!(ty, TyKind::Ref(_, _, _) || is_mutable(&expr) {
   |
```

Instead, this happened:

```log
warning: this boolean expression can be simplified
  --> src/main.rs:32:8
   |
32 |     if !(matches!(ty, TyKind::Ref(_, _, _)) && !is_mutable(&expr)) {
   |        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#nonminimal_bool
   = note: `#[warn(clippy::nonminimal_bool)]` on by default
help: try
   |
32 ~     if !match $expression {
33 +             $pattern $(if $guard)? => true,
34 +             _ => false
35 ~         } || is_mutable(&expr) {
   |
```

### Version

```text
rustc 1.87.0-nightly (00f245915 2025-02-26)
binary: rustc
commit-hash: 00f245915b0c7839d42c26f9628220c4f1b93bf6
commit-date: 2025-02-26
host: x86_64-unknown-linux-gnu
release: 1.87.0-nightly
LLVM version: 20.1.0
```

**Comment by profetia** [comment]
Looks like the cause is that `Sugg` does not use the `source_callsite` on spans. But changing it will cause these lints that intended to work inside macro to break. This is tricky.
